### ISSUE TITLE
Strawberry Shake's generated code does not handle custom output directory properly.

### ISSUE THREAD

**Issue by kellyrbourg** [ChilliCream/graphql-platform#5532]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Our repo is setup as follows. (We are using [NX](https://nx.dev/) for managing build orchestration/targets/dependencies/artifacts). We have a custom output directory set in our root Directory.Build.props that is set to an absolute path. (/home/krbourg/code/alloy/dist)

```
/dist/libs/shared/infrastructure/strawberry-shake-server-client/bin/*
/dist/libs/shared/infrastructure/strawberry-shake-server-client/obj/*
/libs/share/infrastructure/strawberry-shake-server-client
/Directory.Build.props
```

We are seeing the Strawberry Shake code generation output files to the following location:
`/libs/shared/infrastructure/strawberry-shake-server-client/home/krbourg/work/code/alloy/dist/libs/shared/infrastructure/strawberry-shake-server-client`
It seems to be taking our absolute path, and treating it as if it should be relative to the project directory.

### Steps to reproduce

1. Create new dotnet project within a nested folder and add strawberry shake + schema/operations. (We used StrawberryShake.Server & StrawberryShake.Tool)
2. Add a Directory.Build.props file to the root directory and specify an absolute path for the `<BaseIntermediateOutputPath>` property.
3. Build project and observe the code generated relative to the project directory.

### Product

Strawberry Shake

### Version

13.0.0-preview.76

**Comment by michaelstaib** [comment]
Thanks for reporting!

**Comment by AlexandraChibisov** [comment]
Hi, I'd love to take a look at this one! Does anyone have any knowledge at which files/docs I should read up on to start out?

**Comment by michaelstaib** [comment]
@AlexandraChibisov this should be a simple one. I can walk you through in our zoom call.

**Comment by michaelstaib** [comment]
Fixed with #5770
